tools/block-generator/generator/make_transactions.go

```diff
@@ -20,15 +20,34 @@ import (
 	"encoding/binary"
 
 	"github.com/algorand/go-algorand/data/basics"
-	"github.com/algorand/go-algorand/data/transactions"
+	txn "github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/txntest"
 	"github.com/algorand/go-algorand/protocol"
 )
 
-func (g *generator) makeTxnHeader(sender basics.Address, round, intra uint64) transactions.Header {
+// ---- header / boilerplate ----
+
+func (g *generator) makeTxnHeader(sender basics.Address, round, intra uint64) txn.Header {
+	note := make([]byte, 8)
+	binary.LittleEndian.PutUint64(note, uint64(g.txnCounter+intra))
+
+	return txn.Header{
+		Sender:      sender,
+		Fee:         basics.MicroAlgos{Raw: g.params.MinTxnFee},
+		FirstValid:  basics.Round(round),
+		LastValid:   basics.Round(round + 1000),
+		GenesisID:   g.genesisID,
+		GenesisHash: g.genesisHash,
+		Note:        note,
+	}
+}
+
+// makeTestTxn creates and populates the flat txntest.Txn structure with the given values.
+func (g *generator) makeTestTxn(sender basics.Address, round, intra uint64) txntest.Txn {
 	note := make([]byte, 8)
 	binary.LittleEndian.PutUint64(note, uint64(g.txnCounter+intra))
```

Comment:
nit: uint64 casting is not needed; same for makeTxnHeader

Reply:
028e9e88165728a4de35a1e0f0cc974d3e3df435